packages/amplify-codegen-appsync-model-plugin/src/utils/process-connections.ts

```diff
@@ -51,16 +51,14 @@ export function getConnectedField(field: CodeGenField, model: CodeGenModel, conn
   if (connectionFields) {
     let keyDirective;
     if (keyName) {
-      throw new Error('DataStore does not support connection directive with keyName');
-      // When the library starts supporting remove the exception
-      // keyDirective = connectedModel.directives.find(dir => {
-      //   return dir.name === 'key' && dir.arguments.name === keyName;
-      // });
-      // if (!keyDirective) {
-      //   throw new Error(
-      //     `Error processing @connection directive on ${model.name}.${field.name}, @key directive with name ${keyName} was not found in connected model ${connectedModel.name}`,
-      //   );
-      // }
+      keyDirective = connectedModel.directives.find(dir => {
+        return dir.name === 'key' && dir.arguments.name === keyName;
```

Comment:
`dir.arguments` defined always or pre-checked when execution gets here?

Reply:
GraphQL JS ensures that